src/commands/functions-artifacts-setup-cleanup-policy.ts

```diff
@@ -0,0 +1,191 @@
+import * as clc from "colorette";
+
+import { Command } from "../command";
+import { FirebaseError } from "../error";
+import { needProjectId } from "../projectUtils";
+import { confirm } from "../prompt";
+import { requirePermissions } from "../requirePermissions";
+import { requireAuth } from "../requireAuth";
+import { logBullet, logSuccess } from "../utils";
+import * as artifactregistry from "../gcp/artifactregistry";
+import * as artifacts from "../functions/artifacts";
+
+/**
+ * Command to set up a cleanup policy for Cloud Run functions container images in Artifact Registry
+ */
+export const command = new Command("functions:artifacts:setup-cleanup-policy")
+  .description(
+    "Set up a cleanup policy for Cloud Run functions container images in Artifact Registry. " +
+      "This policy will automatically delete old container images created during functions deployment.",
+  )
+  .option(
+    "--location <location>",
+    "Specify location to set up the cleanup policy. " +
+      "If omitted, uses the default functions location.",
+    "us-central1",
+  )
+  .option(
+    "--days <days>",
+    "Number of days to keep container images before deletion. Default is 3 days.",
+    "3",
+  )
+  .option(
+    "--none",
+    "Opt-out from cleanup policy. This will prevent suggestions to set up a cleanup policy during initialization and deployment.",
+  )
+  .withForce("Automatically create or modify cleanup policy")
+  .before(requireAuth)
+  .before(artifactregistry.ensureApiEnabled)
+  .before(requirePermissions, [
+    "artifactregistry.repositories.update",
+    "artifactregistry.versions.delete",
+  ])
+  .action(async (options: any) => {
+    const projectId = needProjectId(options);
+    const location = options.location || "us-central1";
+    const daysToKeep = parseInt(options.days || "5", 10);
+
+    const repoPath = artifacts.makeRepoPath(projectId, location);
+    let repository: artifactregistry.Repository;
+    try {
+      repository = await artifactregistry.getRepository(repoPath);
+    } catch (err: any) {
+      if (err.status === 404) {
+        logBullet(`Repository '${repoPath}' does not exist in Artifact Registry.`);
+        logBullet(
+          `Please deploy your functions first using: ` +
+            `${clc.bold(`firebase deploy --only functions`)}`,
+        );
+        return;
+      }
+      throw err;
+    }
+
+    if (options.none) {
+      if (artifacts.hasCleanupOptOut(repository)) {
```

Comment:
I wonder if we should also check that there is no policy. A customer may have opted out of the policy before, set a policy in the Cloud Console, and then assumed the CLI would be able to modify it.

Reply:
good call. fixed.